### PR TITLE
Login/Signup Pages

### PR DESCRIPTION
**Pull request by hodge-jai**
Note: The large diff mostly comes from changes to package-lock.json that occurred when running npm install.

### What this PR does (required):
- brings login and sign up pages in line with the mock ups

### Screenshots / Videos (front-end only):
- Post at least one screenshot to show the feature in action. **Only required in front-end work**

Screen Shot 2021-11-08 at 6 58 10 PM

Screen Shot 2021-11-08 at 6 57 38 PM

### Any information needed to test this feature (required):
- Need to test the responsiveness of the page for different screen sizes

### CODE REVIEW

File: client/src/components/AuthHeader/useStyles.ts [hunk 1]
```diff
@@ -2,15 +2,15 @@ import { makeStyles } from '@material-ui/core/styles';
 
 const useStyles = makeStyles((theme) => ({
   authHeader: {
-    alignSelf: 'flex-end',
-    justifyContent: 'flex-end',
+    alignSelf: 'center',
```
Comment (rajivtitus) [inline client/src/components/AuthHeader/useStyles.ts]: I know this is a part of the starter code but most of these styles can be applied to the Box component directly (it is a good practice to pass props to MUI components unless we explicitly need a class to apply some styles) 😄 

Eg: `<Box display="flex" alignItems="center" justifyContent="center"...>`

File: client/src/components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export default function NavBar(): JSX.Element {
+  const classes = useStyles();
+
+  return (
+    <AppBar position="static">
+      <ToolBar className={classes.toolbar}>
+        <img src={logo} alt="logo" />
+        <Grid className={classes.loginsignup}>
+          <Box p={1}>
+            <Link to="/login" className={classes.link}>
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: You can combine the MUI button with react-router's Link :)

Eg: 
```
<Button component={Link} to="/about" variant="contained" color="primary">
  About Page
</Button>
```

File: client/src/components/NavBar/NavBar.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+              <Button color="primary" size="large" variant="outlined">
+                LOGIN
+              </Button>
+            </Link>
+          </Box>
+          <Box p={1}>
+            <Link to="signup" className={classes.link}>
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: Ditto as comment above!

File: client/src/components/NavBar/NavBar.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+  return (
+    <AppBar position="static">
+      <ToolBar className={classes.toolbar}>
+        <img src={logo} alt="logo" />
+        <Grid className={classes.loginsignup}>
+          <Box p={1}>
+            <Link to="/login" className={classes.link}>
+              <Button color="primary" size="large" variant="outlined">
+                LOGIN
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx]: Please refrain from using CAPITALCASE text as some screen readers interpret them as abbreviations, you can use a CSS class to transform text instead.

File: client/src/components/NavBar/NavBar.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import Grid from '@material-ui/core/Grid';
+import AppBar from '@material-ui/core/AppBar';
+import ToolBar from '@material-ui/core/Toolbar';
+import Button from '@material-ui/core/Button';
+import Box from '@material-ui/core/Box';
+import { Link } from 'react-router-dom';
+import useStyles from './useStyles';
+import logo from '../../Images/logo.png';
+
+export default function NavBar(): JSX.Element {
```
Comment (rajivtitus) [inline client/src/components/NavBar/NavBar.tsx:10]: The NavBar is a separate feature, good work completing it within this PR. However, in the future please make sure you create a new PR for each feature :)

File: client/src/components/NavBar/useStyles.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import { makeStyles } from '@material-ui/core/styles';
+
+const useStyles = makeStyles(() => ({
+  toolbar: {
+    display: 'flex',
+    justifyContent: 'space-between',
+    minHeight: 56,
+    backgroundColor: '#FFFFFF',
+  },
+  loginsignup: {
```
Comment (rajivtitus) [inline client/src/components/NavBar/useStyles.ts]:
```suggestion
  loginSignup: {
```
Nitpick: Please make sure you use camelCase when writing CSS in JS 😄

File: client/src/themes/theme.ts [hunk 7]
```diff
@@ -2,15 +2,19 @@ import { createMuiTheme } from '@material-ui/core';
 
 export const theme = createMuiTheme({
   typography: {
-    fontFamily: '"Open Sans", "sans-serif", "Roboto"',
+    fontFamily: '"Roboto"',
```
Comment (rajivtitus) [inline client/src/themes/theme.ts:5]: Nice work updating the styles within the theme 💯

File: client/src/pages/SignUp/SignUpForm/useStyles.ts [hunk 8]
```diff
@@ -17,7 +17,7 @@ const useStyles = makeStyles((theme) => ({
   },
   forgot: {
     paddingRight: 10,
-    color: '#3a8dff',
+    color: '#f14140',
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/useStyles.ts]: You can add these colours to a palette within MUI's theme and then use the theme to apply them to ensure consistency across the app!

File: client/src/pages/SignUp/useStyles.ts [hunk 9]
```diff
@@ -6,21 +6,23 @@ const useStyles = makeStyles(() => ({
...
-    paddingTop: 23,
+    minHeight: '100%',
   },
   welcome: {
     fontSize: 26,
     paddingBottom: 20,
     color: '#000000',
     fontWeight: 700,
-    fontFamily: "'Open Sans'",
+    fontFamily: "'Roboto'",
```
Comment (rajivtitus) [inline client/src/pages/SignUp/useStyles.ts]: Please avoid setting the font family outside of MUI's theme, you can use the theme to apply it instead!

File: client/src/pages/SignUp/SignUpForm/SignUpForm.tsx [hunk 10]
```diff
@@ -109,9 +113,10 @@ const SignUpForm = ({ handleSubmit }: Props): JSX.Element => {
 
           <Box textAlign="center">
             <Button type="submit" size="large" variant="contained" color="primary" className={classes.submit}>
-              {isSubmitting ? <CircularProgress style={{ color: 'white' }} /> : 'Create'}
+              {isSubmitting ? <CircularProgress style={{ color: 'white' }} /> : 'SIGN UP'}
             </Button>
           </Box>
+          <AuthHeader linkTo="/login" asideText="Already have an account?" btnText="Login" />
```
Comment (rajivtitus) [inline client/src/pages/SignUp/SignUpForm/SignUpForm.tsx]: Optional: Maybe we can rename this component? Calling it AuthHeader doesn't seem appropriate 😄

File: client/src/components/AuthHeader/useStyles.ts (hunk 1 above)
Comment (hodge-jai) [reply client/src/components/AuthHeader/useStyles.ts]: Changes have been made. Can I go ahead and merge this?

Comment (rajivtitus) [reply client/src/components/AuthHeader/useStyles.ts]: Please wait for approval from a Hatchways team member prior to merging PRs :)

**Comment by rajivtitus** [comment]
@hodge-jai Please address the pending comments and tag me in for another review once you are done :)

**Comment by hodge-jai** [comment]
@rajivtitus Changes have been made. Please review again!

**Review by rajivtitus — COMMENTED** [review]
Reposting the additional comments. Please have a look and address :)

**Comment by hodge-jai** [comment]
@rajivtitus Please take a look at my changes and review!

**Review by rajivtitus — APPROVED** [review]
Excellent work making those changes! 💯

**Pull request merged**